python/ray/dag/compiled_dag_node.py

```diff
@@ -1143,11 +1179,12 @@ def _get_or_compile(
                     assert isinstance(reader_task.dag_node, ClassMethodNode)
                     reader_handle = reader_task.dag_node._get_actor_handle()
                     reader_node_id = self._get_node_id(reader_handle)
-                    for arg in reader_task.args:
-                        if isinstance(arg, InputAttributeNode) or isinstance(
-                            arg, InputNode
+                    for upstream_node in reader_task.dag_node._upstream_nodes:
```

Comment:
If a reader puts `InputNode` in a list, the `InputNode` will not be added to `input_node_to_reader_and_node_set`.